backend/app/src/main/java/ch/zhaw/studyflow/controllers/DegreeController.java

```diff
@@ -0,0 +1,144 @@
+        this.degreeManager = degreeManager;
+    }
+
+
+    @Endpoint(method = HttpMethod.POST)
+    public HttpResponse createDegree(RequestContext requestContext) {
+        final HttpRequest request = requestContext.getRequest();
+
+        return authenticationHandler.handleIfAuthenticated(request, principal -> {
+            final HttpResponse response = request.createResponse()
+                    .setStatusCode(HttpStatusCode.BAD_REQUEST);
+
+            principal.getClaim(CommonClaims.USER_ID)
+                    .flatMap(userId -> request.getRequestBody()
+                            .flatMap(body -> body.tryRead(Degree.class))
+                            .map(degree -> {
+                                degreeManager.createDegree(userId, degree);
+                                return degree;
+                            }))
+                    .ifPresent(value ->
+                            response.setResponseBody(JsonContent.writableOf(value))
+                                    .setStatusCode(HttpStatusCode.CREATED)
+                    );
+            return response;
+        });
+    }
+
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getDegrees(RequestContext requestContext) {
+        final HttpRequest request = requestContext.getRequest();
+
+        return authenticationHandler.handleIfAuthenticated(request, principal -> {
+            final HttpResponse response = request.createResponse()
+                    .setStatusCode(HttpStatusCode.BAD_REQUEST);
+
+            principal.getClaim(CommonClaims.USER_ID)
+                    .ifPresent(userId ->
+                            response.setResponseBody(JsonContent.writableOf(degreeManager.getDegreesForStudent(userId)))
+                                    .setStatusCode(HttpStatusCode.OK)
+                    );
+            return response;
+        });
+    }
+
+    @Route(path = "{degreeId}")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getDegree(RequestContext requestContext) {
+        final HttpRequest request = requestContext.getRequest();
+
+        return authenticationHandler.handleIfAuthenticated(request, principal -> {
+            final HttpResponse response = request.createResponse()
+                    .setStatusCode(HttpStatusCode.BAD_REQUEST);
+
+            final Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID);
+            final Optional<Long> degreeId = requestContext.getUrlCaptures().get("degreeId").flatMap(LongUtils::tryParseLong);
+            if (userId.isPresent() && degreeId.isPresent()) {
+                final Degree requestedDegree = degreeManager.getDegree(degreeId.get());
+                if (requestedDegree != null) {
+                    response.setResponseBody(JsonContent.writableOf(requestedDegree))
+                            .setStatusCode(HttpStatusCode.OK);
+                }
+            }
+            return response;
+        });
+    }
+
+    @Route(path = "{degreeId}")
```

Comment:
The PATCH function still seems to make some problems, i have noticed that in other Controllers it's also still not working.
For example in the ModulesController, the modules gets updated correctly but im getting no response and it's like stuck in a loop.

And Here i also now tested the PATCH function to see if it acts the same and it does.
What can be the issue ?